### PR TITLE
chore: dep upgrade: read-pkg-up

### PR DESCRIPTION
**Pull request by aaronArinder**
Edit: read-pkg-up ^8.0.0 doesn't like earlier node versions, so going to use circle to test out different versions and see if I can't get those tests passing. Ignore this for now.

### What does this PR do?
Upgrades `read-pkg-up` in response to [Snyk flagging one of it's sub-dependencies for regex dos](https://snyk.io/test/npm/dd-trace/0.30.6). Note that the version in that Snyk advisory is an earlier version of dd-trace; but, it should be flagging the same advisory for the current version of dd-trace (and, at any rate, it _is_ in a private work repo).

### Motivation
Snyk flagged this in a private repo and it seemed like a fairly straightforward change. The `read-pkg-up` API only changes [slightly](https://www.npmjs.com/package/read-pkg-up).

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Additional Notes
I couldn't run the tests for this. This is what `yarn test` (after spinning things up with docker compose) spits out a bunch of errors from a node native addon, couchbase. Here's a snippet from the end to show the version/culprit, but I can send over the full list if you want:

```
fatal error: too many errors emitted, stopping now [-ferror-limit=]
16 warnings and 20 errors generated.
make: *** [Release/obj.target/couchbase_impl/src/couchbase_impl.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/aaronarinder/.nvm/versions/node/v12.18.3/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/Users/aaronarinder/.nvm/versions/node/v12.18.3/bin/node" "/Users/aaronarinder/.nvm/versions/node/v12.18.3/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/aaronarinder/external-code/pull-requests/dd-trace-js/versions/couchbase@2.4.2/node_modules/couchbase
```

The circleci cli spits this out:

```
Error:
CircleCI was unable to start the container because the container entrypoint or command failed to start.

This typically means that the passed entrypoint or command is not found or is not valid.  Try clearing the entrypoint/command values.
Original error: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: process_linux.go:459: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: time="2021-03-30T16:52:09Z" level=fatal msg="no such file or directory": unknown

Step failed
Task failed
```

### Additional _Additional_ Notes
DataDog is awesome! We're enjoying the service where I work, to the point where I can't imagine doing without it. So, thanks!

### CODE REVIEW

File: packages/dd-trace/src/pkg.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+const { readPackageUpSync } = require('read-pkg-up')
 
 function findRoot () {
   return require.main && require.main.filename ? path.dirname(require.main.filename) : process.cwd()
 }
 
 function findPkg () {
   const cwd = findRoot()
-  const up = readPkgUp.sync({ cwd })
+  const up = readPackageupSync({ cwd })
```
[the remark this answers is not in the document]
Comment (aaronArinder) [reply packages/dd-trace/src/pkg.js]: gosh dern it

**Comment by aaronArinder** [comment]
Leaving this open for the maintainers to get a second set of eyes on, but tests are failing:

```
  1) pkg
       should load the service name from the main module:
     AssertionError: expected undefined to equal 'mocha'
      at Context.it (packages/dd-trace/test/pkg.spec.js:27:25)
      at scope.activate (packages/dd-trace/src/scope/base.js:1:20081)
      at Scope._activate (packages/dd-trace/src/scope/async_hooks.js:3:1410)
      at Scope.activate (packages/dd-trace/src/scope/base.js:1:18663)
      at Context.bound (packages/dd-trace/src/scope/base.js:1:19999)

  2) pkg
       should load the version number from the main module:
     AssertionError: expected undefined to match /^\d+.\d+.\d+/
      at Context.it (packages/dd-trace/test/pkg.spec.js:31:28)
      at scope.activate (packages/dd-trace/src/scope/base.js:1:20081)
      at Scope._activate (packages/dd-trace/src/scope/async_hooks.js:3:1410)
      at Scope.activate (packages/dd-trace/src/scope/base.js:1:18663)
      at Context.bound (packages/dd-trace/src/scope/base.js:1:19999)

  3) pkg
       in pre-require
         should load the package.json correctly:
     AssertionError: expected undefined to equal 'dd-trace'
      at Context.it (packages/dd-trace/test/pkg.spec.js:17:29)
      at scope.activate (packages/dd-trace/src/scope/base.js:1:20081)
      at Scope._activate (packages/dd-trace/src/scope/async_hooks.js:3:1410)
      at Scope.activate (packages/dd-trace/src/scope/base.js:1:18663)
      at Context.bound (packages/dd-trace/src/scope/base.js:1:19999)
      ```
      
     
     I gave major versions 8, 7, and 6 a shot, with 8 requiring too high of a node version, and with all three failing the tests above. It's too annoying to troubleshoot without being able to run the tests locally (which is likely just something up with my machine), so I'm just leaving it off here.

**Comment by rochdev** [comment]
Closing as we removed `read-pkg-up` completely in 0.32.2.

**Comment by aaronArinder** [comment]
nice! thank you!

**Pull request closed without merging**